File: gradle-java-server/src/main/java/com/github/genraven/gradlejavaserver/service/SpellService.java
package com.github.genraven.gradlejavaserver.service;

import com.github.genraven.gradlejavaserver.domain.spell.Spell;
import com.github.genraven.gradlejavaserver.repository.SpellRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Service
public class SpellService {
    private final SpellRepository spellRepository;

    @Autowired
    public SpellService(final SpellRepository spellRepository) {
        this.spellRepository = spellRepository;
    }

    public Flux<Spell> getAllSpells() {
        return spellRepository.findAll();
    }

    public Mono<Spell> getSpell(final String name) {
        return spellRepository.findById(name);
    }

    public Mono<Spell> createSpell(final String name) {
        return spellRepository.save(new Spell(name));
    }

    public Mono<Spell> updateSpell(final String name, final Spell spell) {
        return getSpell(name).map(sp -> {
            sp.setDescription(spell.getDescription());
            sp.setConcentration(spell.isConcentration());
            sp.setDifficulty(spell.getDifficulty());
            sp.setSkills(spell.getSkills());
            sp.setEffects(spell.getEffects());
            return sp;
        }).flatMap(spellRepository::save);
    }
}

File: src/services/SpellService.ts
import axios from "axios";
import {RootPath} from "./RootPath";
import Spell from "../models/spell/Spell";

export default class SpellService {

    static async getSpells(): Promise<Spell[]> {
        return await (await axios.get(RootPath.Spell)).data;
    }

    static async getSpell(name: string): Promise<Spell> {
        return await (await axios.get(RootPath.Spell + name)).data;
    }

    static async createSpell(name: string): Promise<Spell> {
        return await (await axios.post(RootPath.Spell + name)).data;
    }

    static async updateSpell(name: string, spell: Spell): Promise<Spell> {
        return await (await axios.put(RootPath.Spell + name, spell)).data;
    }
}